stubs/CoreGenericIterators.phpstub

```diff
@@ -319,8 +319,9 @@ class ArrayIterator implements SeekableIterator, ArrayAccess, Serializable, Coun
 /**
  * @template-covariant TKey
  * @template-covariant TValue
+ * @template TIterator as Traversable<TKey, TValue>
```

Comment:
adding a new template to a already existing type is kind-of a BC break, as all existing code, which didn't error before this PR will now start erroring about a missing template.. 

thats why phpstan ignores the generic types when newly added like https://github.com/phpstan/phpstan-src/pull/796/files#diff-e4b571e8dcdf42219808e950c92c6335fe4e140e4b0b2d01bc5b41b2f25f717eR25-R28

does psalm have similar mechanics?

Reply:
Well this would qualify as a bug in inferrence, which is not covered by our BC promise IMHO.

Anyway, this is currently targeted on master (so Psalm 5) so this would not be an issue.

I just have to spend time on this, those are classes I never use so I have some difficulties seeing how it should be done